### PR TITLE
DAOS-14798 test: Fix key lookups on changed dmg JSON for SmdDevices

### PR DESCRIPTION
**Pull request by tanabarr**
Features: control pool
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/checksum/csum_error_logging.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def get_checksum_error_value(self, dmg, device_id):
             for device in devices:
                 try:
                     if device['uuid'] == device_id:
-                        return device['health']['checksum_errs']
+                        return device['ctrlr']['health_stats']['checksum_errs']
```
Comment (daltonbohning) [inline src/tests/ftest/checksum/csum_error_logging.py:40]: Unfortunately, this test isn't covered by `Features: control pool`

You can use `Features: control pool CsumErrorLog`

Or, if you think it should be covered under `control` or `pool`, you can add that tag to the third line of tags here:
https://github.com/daos-stack/daos/blob/bcdc5487c7f9ce1a37fb92ab051a6a8e8c21f5dd/src/tests/ftest/checksum/csum_error_logging.py#L55

File: src/tests/ftest/util/soak_utils.py [hunk 2]
```diff
@@ -427,8 +427,8 @@ def launch_vmd_identify_check(self, name, results, args):
         for value in list(result['response']['host_storage_map'].values()):
             if value['storage']['smd_info']['devices']:
                 for device in value['storage']['smd_info']['devices']:
-                    if device['led_state'] != "QUICK_BLINK":
-                        failing_vmd.append([device['tr_addr'], value['hosts']])
+                    if device['ctrlr']['led_state'] != "QUICK_BLINK":
+                        failing_vmd.append([device['ctrlr']['pci_addr'], value['hosts']])
```
Comment (daltonbohning) [inline src/tests/ftest/util/soak_utils.py:431]: @mjean308 FYI on this one. Not sure if soak_smoke would cover this

File: src/tests/ftest/checksum/csum_error_logging.py (hunk 1 above)
Comment (tanabarr) [reply src/tests/ftest/checksum/csum_error_logging.py:40]: Currently running with `Test-tag: vmd,vmd_led nvme,pool checksum,faults,daos_test pool,pool_create_all` which should cover most of the fixes.

File: src/tests/ftest/util/soak_utils.py (hunk 2 above)
Comment (phender) [reply src/tests/ftest/util/soak_utils.py:431]: That is only executed by the `test_soak_online_harassers` and that particular harasser - `vmd-identify-check` - is commented out currently.

**Review by phender — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
Of the 4 failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13507/2/testReport/, 3 are from manual tagged tests and the one daily test is https://daosio.atlassian.net/browse/DAOS-14801

**Pull request merged**